packages/react-storage/src/components/StorageBrowser/controls/hooks/useActionStart.tsx

```diff
@@ -0,0 +1,28 @@
+import { ActionStartProps } from '../../composables/ActionStart';
+import { useControlsContext } from '../../controls/context';
+
+export type UseActionStart = () => {
+  props?: ActionStartProps;
+};
+
+export const useActionStart: UseActionStart = () => {
+  const { data, actionsConfig } = useControlsContext();
+  const { taskCounts } = data;
+  let disabled = actionsConfig.actionStart?.disabled;
+  if (disabled === undefined && taskCounts) {
+    const hasStarted = !!taskCounts.PENDING;
+    const hasCompleted =
+      !!taskCounts.TOTAL &&
+      taskCounts.CANCELED + taskCounts.COMPLETE + taskCounts.FAILED ===
+        taskCounts.TOTAL;
+
+    disabled = !taskCounts.TOTAL || hasStarted || hasCompleted;
+  }
```

Comment:
Hmm. I think this control (and hook) should really either fully encapsulate the logic for determining what it means for it to be "disabled" or not at all. Imo, it should probably be the latter since the logic for determining this could vary from action to action. I think we should probably move lines 12-20 back into the UploadControls for now.